### ISSUE TITLE
distsql: many nightly logic tests failing since 2 days ago

### ISSUE THREAD

**Issue by jordanlewis** [cockroachdb/cockroach#24501]
https://teamcity.cockroachdb.com//viewLog.html?buildId=591761&buildTypeId=Cockroach_SqlLogicTests&tab=buildLog&_focus=7470

```
E180405 06:32:10.815810 15458 sql/distsqlrun/server.go:375  [n2] error setting up flow: "$4" >= 48:::INT: unsupported comparison operator: <string> >= <int>
E180405 06:32:10.815932 15274 sql/distsqlrun/server.go:375  [n3] error setting up flow: "$4" >= 48:::INT: unsupported comparison operator: <string> >= <int>
E180405 06:32:10.816378 85 sql/logictest/logic.go:2112  
/go/src/github.com/cockroachdb/sqllogictest/test/index/commute/10/slt_good_19.test:296: SELECT pk FROM tab1 WHERE col0 > 27 AND col3 >= 48
expected success, but found
pq: unsupported comparison operator: <string> >= <int>
```

This looks related to #24401, but I suspect #24373 as its the only distsql-related patch that went in between the last successful run and the first failed run.

**Comment by andreimatei** [comment]
Seems to be this commit: https://github.com/cockroachdb/cockroach/commit/ba2aee3ec01e8ea8b9340b47d2430f70d4ffc7d6#diff-88ff3cb5970889e8ef59e5a9aeb35dff

Will revert it if I don't see the problem soon.

**Comment by andreimatei** [comment]
@pbardea this has been fixed by #24697 , right? I'm running one of the nightlies for a few minutes now and it seems happy so far.

**Comment by pbardea** [comment]
@andreimatei  Yep should be fixed.
